edb/repl/__init__.py

```diff
@@ -71,7 +73,17 @@ def is_multiline():
         return False
 
     if text.endswith(';'):
-        return not lexutils.split_edgeql(text)
+        try:
+            return lexutils.split_edgeql(text) is None
```

Comment:
I find this interface to be quite convoluted.  Why can't this logic be put _inside_ the `split_edgeql()` function and have it return both the list of statements _and_ whether it thinks the last statement has been terminated?  Perhaps as a `(complete, incomplete)` tuple.  That way the repl can execute the `complete` sequence, and populate the prompt with the text from `incomplete`:

```
db> SELECT 1; SELECT 2 <alt+enter>
{1}
db>SELECT 2█
```

Reply:
Also, this approach removes the need for the `script_mode` argument, since the decision whether to execute the `incomplete` part would fall onto the caller.